staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -941,6 +969,13 @@ var (
 	}
 )
 
+func getNodePredicatesForService(service *v1.Service) []NodeConditionPredicate {
+	if service.Spec.ExternalTrafficPolicy == v1.ServiceExternalTrafficPolicyTypeLocal {
+		return []NodeConditionPredicate{nodeIncludedPredicate, nodeSchedulablePredicate, nodeUnTaintedPredicate}
```

Comment:
make another var `etpLocalNodePredicates` or move `allNodePredicates` into this function